File: src/main/kotlin/dev/jefy/connectpro/engagement/applicaion/dtos/ReviewResponse.java
package dev.jefy.connectpro.engagement.applicaion.dtos;

import java.time.Instant;
import java.util.UUID;

import dev.jefy.connectpro.engagement.domain.Review;

/**
 * @author Jôph Yamba
 */
public record ReviewResponse(
        UUID reviewerId,
        UUID serviceId,
        int rating,
        String comment,
        Instant createdAt
) {
    public static ReviewResponse fromDomain(Review review) {
        return new ReviewResponse(
                review.getId().serviceId(),
                review.getId().reviewerId(),
                review.getRating().value(),
                review.getComment(),
                review.getCreatedAt()
        );
    }
}

File: src/main/kotlin/dev/jefy/connectpro/engagement/application/dtos/ReviewResponse.kt
package dev.jefy.connectpro.engagement.application.dtos

import dev.jefy.connectpro.engagement.domain.Review
import dev.jefy.connectpro.user.application.dtos.UserData
import java.time.Instant
import java.util.*

/**
 * @author Jôph Yamba
 */

data class ReviewResponse(
    val reviewerId: UUID,
    val reviewerName: String,
    val reviewerAvatar: String?,
    val serviceId: UUID,
    val rating: Int,
    val comment: String,
    val createdAt: Instant
)

fun Review.toResponse(reviewer: UserData): ReviewResponse = ReviewResponse(
    serviceId = this.id.serviceId,
    reviewerId = this.id.reviewerId,
    rating = this.rating.value,
    comment = this.comment,
    createdAt = this.createdAt,
    reviewerName = reviewer.name,
    reviewerAvatar = reviewer.imageUrl
)
